### DISCUSSION TITLE
Problems while compiling MOOSE

### DISCUSSION THREAD

**Discussion by Giovar6** [idaholab/moose#16151] [Q&A Getting Started] [answered]
Good evening everybody,

I followed literally the instructions reported at https://mooseframework.inl.gov/getting_started/installation/conda.html, to install MOOSE on Ubuntu 20.04.1 but when I compile MOOSE I got this error:

Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/loops_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/kernels_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/mesh_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/interfacekernels_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/interfaces_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/timesteppers_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/problems_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/systems_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/predictors_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/fvbcs_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/postprocessors_Unity.C...
Compiling C++ (in opt mode) /home/giovanni/projects/moose/framework/build/unity_src/variables_Unity.C...
{standard input}: Assembler messages:
{standard input}:19171: Warning: end of file not at end of a line; newline inserted
x86_64-conda_cos6-linux-gnu-c++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://gcc.gnu.org/bugs/> for instructions.
make: *** [/home/giovanni/projects/moose/framework/build.mk:118: /home/giovanni/projects/moose/framework/build/unity_src/problems_Unity.x86_64-pc-linux-gnu.opt.lo] Errore 1
make: *** Attesa per i processi non terminati....

I've tried also to run the tests, but obviously all tests failed. So I've tried to uninstall everything and reinstall but did not worked. Sometimes also Ubuntu crashed. When this problem happen generally the computer stop working for some minutes (I cannot move neither the mouse)
What could be the reason for this issue?

Thank you in advance for any suggestion.
Kind regards,
Giovanni

**Comment by Giovar6** [comment, marked as answer]
Good Evevening,

The problem was solved. Jeson Miller explained me (at this discussion https://groups.google.com/g/moose-users/c/UjkErXUTfD0) that 6 Gb of RAM are not enough to perform the command make -j4.
I've used instead the command: make -j3 and MOOSE compiled perfectly.
I have also performed the test obtaining several pass and some skip.

Thank you anyway.
Giovanni

**Comment by milljm** [comment]
@Giovar6  Thank you for re-submitting your question!
